snooty/specparser.py

```diff
@@ -262,6 +262,46 @@ def create_role(self) -> Role:
         )
 
 
+@checked
+@dataclass
+class FacetDefinition:
+    name: str
+
+
+@checked
+@dataclass
+class TargetPlatformDefinition:
+    name: str
+    versions: List[FacetDefinition]
+
+@checked
+@dataclass
+class TaxonomySpec:
+
+    genres: List[FacetDefinition]
+    target_platforms: List[TargetPlatformDefinition]
+
+    @classmethod
+    def get_taxonomy(cls) -> TaxonomySpec:
```

Comment:
Right now this could get called quite a bit. I'd rather load the TaxonomySpec once and cache it. See what we do for Spec:

https://github.com/mongodb/snooty-parser/blob/master/snooty/specparser.py#L405-L413

Reply:
done! not as explicit (no need to call anything to load) but it will hold in memory 👍